### PR TITLE
Add metadata for GHC 9.8.1-alpha1

### PR DESCRIPTION
**Pull request by bgamari**
As it says on the tin

### CODE REVIEW

**Comment by bgamari** [comment]
@hasufell, would you mind if I merge this?

**Comment by hasufell** [comment]
> @hasufell, would you mind if I merge this?
> 

Can. Please sign the metadata as well and don't use Github UI to merge.

File: ghcup-prereleases-0.0.7.yaml [hunk 1]
```diff
@@ -837,6 +837,121 @@ ghcupDownloads:
...
+            unknown_versioning: *ghc-981alpha1-deb10
+          Linux_UnknownLinux:
+            unknown_versioning: *ghc-981alpha1-deb10
+        A_64:
+          Darwin:
+            unknown_versioning:
+              dlHash: 54bfebe74a76cbec360cf0f8d84d8f7658b44af3f8df19fddd4fd0ade6f10909
+              dlOutput: ghc-9.8.0.20230727-x86_64-darwin.tar.xz
+              dlSubdir: ghc-9.8.0.20230727-x86_64-apple-darwin
+              dlUri: https://downloads.haskell.org/~ghc/9.8.0.20230727/ghc-9.8.0.20230727-x86_64-apple-darwin.tar.xz
```
Comment (andreasabel) [inline ghcup-prereleases-0.0.7.yaml:861]: This URL does not exist.
@bgamari See
- #102

**Comment by phadej** [comment]
@arybczak 

Do you mean what's printed by `ghc --numeric-version`?

**Comment by phadej** [comment]
> Sigh, quite right. I am rather unsure of how best to handle this. Renaming the alpha is likely to do more harm than good at this point. I

Make alpha2 right. As a `haskell-ci` maintainer I can pretend alpha1 didn't happen.

**Comment by hasufell** [comment]
> Sigh, quite right. I am rather unsure of how best to handle this. Renaming the alpha is likely to do more harm than good at this point.

- https://gitlab.haskell.org/haskell/ghcup-hs/-/issues/73#note_375245
- https://github.com/haskell/cabal/commit/f78d2aaaa76b527f4f03032f256cf183c4ef8f5d
- https://gitlab.haskell.org/ghc/ghc/-/merge_requests/4214

**Comment by andreasabel** [comment]
@bgamari wrote:
> > FYI this uses different versioning scheme than previous alpha releases (`X.Y.1-alphaN` instead of `X.Y.0.DATE`) , which breaks `haskell-ci` interoperability.
> Sigh, quite right. I am rather unsure of how best to handle this. Renaming the alpha is likely to do more harm than good at this point. It would be nice if we could declare version aliases in GHCup metadata.

How about just copying the metadata of 9.8.1-alpha1 to 9.8.0.20230727 so that both are available?  I am used from GHC 9.6.0 already that the preleases go by two different names (0.DATE and 1-alphaN).
Copying instead of aliasing might be less pretty, but would be a pragmatic solution, wouldn't it?  Prereleases are for the experts only anyway, so no need to worry about the masses...

**Pull request merged**